src/cunumeric/sort/sort_cpu.inl

```diff
@@ -0,0 +1,1027 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+// Useful for IDEs
+#include "cunumeric/sort/sort.h"
+#include "cunumeric/pitches.h"
+#include "core/comm/coll.h"
+
+#include <thrust/sort.h>
+#include <thrust/detail/config.h>
+#include <thrust/execution_policy.h>
+#include <thrust/system/omp/execution_policy.h>
+#include <thrust/sequence.h>
+#include <thrust/iterator/constant_iterator.h>
+
+#include <functional>
+#include <numeric>
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+// sorts inptr in-place, if argptr not nullptr it returns sort indices
+template <typename VAL, typename DerivedPolicy>
+void thrust_local_sort_inplace(VAL* inptr,
+                               int64_t* argptr,
+                               const size_t volume,
+                               const size_t sort_dim_size,
+                               const bool stable_argsort,
+                               const thrust::detail::execution_policy_base<DerivedPolicy>& exec)
```

Comment:
you could make the whole type a template parameter instead (i.e., `const DerivedPolicy& exec` would work too), as detail types like this are subject to change.

Reply:
Good hint, I was not aware of this.